back/authentication.ts

```diff
@@ -79,19 +82,20 @@ export class auth_module {
      * Checks if username is valid.
      * Should contain [3, 16] symbols.
      * First symbol is a-z | A-Z
-     * Others a-z | A-Z | 0-9
+     * Others a-z | A-Z | 0-9 | _
      *
      * @param {string} username
      * @return {*} boolean
      */
     is_username_valid ( username: string ): boolean {
-        const regex = /[a-zA-Z]([a-zA-Z0-9]{2,15})/;
+        const regex = /^[a-zA-Z]([a-zA-Z0-9_]{2,15})$/;
```

Comment:
Great job! But we must be sure, that in our project there are no place for abusive and offensive names. For example "nigger", "fuck", "jirnaya_femka" and especially "Winnie the Pooh" for case if our service will work on Chinese market. Thank you

Reply:
I am not sure that such checks are necessary since we are not planning to deploy this application in places where there is censorship on certain words in usernames. Tbh. most participants don't want this project to be more than something that will give them a pass on the exam